syntax = "proto3";

service DataWarehouseService {
  rpc sendWarehouse (Warehouse) returns (Ack);
}

message Warehouse {
  string warehouseID = 1;
  string warehouseName = 2;
  string warehouseAddress = 3;
  string warehousePostalCode = 4;
  string warehouseCity = 5;
  string warehouseCountry = 6;
  string timestamp = 7;
  repeated ProductData productdata = 8;
}

message ProductData {
  string productID = 1;
  string productName = 2;
  string productCategory = 3;
  int32 productQuantity = 4;
  string productUnit = 5;
}

message Ack {
  string message = 1;
}
